packages/components/src/font-size-picker/index.js

```diff
@@ -51,38 +57,40 @@ function FontSizePicker( {
 		onChange( Number( newValue ) );
 	};
 
-	const onSelectChangeValue = ( eventValue ) => {
-		setCurrentSelectValue( eventValue );
-		const selectedFont = fontSizes.find( ( font ) => font.slug === eventValue );
-		if ( selectedFont ) {
-			onChange( selectedFont.size );
-		}
+	const onSelectChangeValue = ( { selectedItem } ) => {
+		setCurrentSelectValue( selectedItem.key );
+		onChange( selectedItem.style && selectedItem.style.fontSize );
 	};
 
+	const items = getSelectOptions( fontSizes );
+	const rangeControlNumberId = `components-range-control__number#${ instanceId }`;
 	return (
 		<fieldset className="components-font-size-picker">
-			<legend>
+			<legend className="screen-reader-text">
 				{ __( 'Font Size' ) }
 			</legend>
 			<div className="components-font-size-picker__controls">
 				{ ( fontSizes.length > 0 ) &&
-					<SelectControl
+					<CustomSelect
 						className={ 'components-font-size-picker__select' }
-						label={ 'Choose preset' }
-						hideLabelFromVision={ true }
-						value={ currentSelectValue }
-						onChange={ onSelectChangeValue }
-						options={ getSelectOptions( fontSizes ) }
+						label={ __( 'Preset Size' ) }
+						items={ items }
+						selectedItem={ items.find( ( item ) => item.key === currentSelectValue ) || items[ 0 ] }
+						onSelectedItemChange={ onSelectChangeValue }
```

Comment:
Why not just onChange (I'm trying to see if we can mirror the SelectControl API props, so it's consistent and switching between the two becomes easy.

Reply:
I tried to mirror the `Downshift` props.

Their reasoning is probably that you also have `highlightedItem`, `onHighlightedItemChange`, etc. They prefer to be explicit about the two instead of having `selectedItem == value`.

Reply:
I have a preference for staying consistent across our components, Downshift is just an implementation detail that can be changed at any time.

Reply:
I think that having `onChange` and `onHighlightedValueChange` is a bit confusing. It's not totally clear that `onChange` does not include highlighted value changes.

Also, `selectedItem` takes the whole object, not just the key.

Similarly, `onSelectedItemChange` gives you the whole object, not just the key.

This already makes them fundamentally different to `SelectControl`'s `value` and `onChange`, but they are a much simpler to use API as you don't have to handle all of the key-to-object normalization/searching on the consumer side.

Reply:
I feel like highlighting is not something you'd need to care about often as a consumer, I'm not sure we even need it as an API in our component to be honest.

Also, I just noticed that the component is lacking a README.

Reply:
We already have the functionality for free and this is a reusable component meant to be used outside of Gutenberg as well so I don't see how we gain from not exposing it.

Also, `selectedItem` and `items` very clearly convey the contract that the `selectedItem` should be an object from the `items` list.

>Also, I just noticed that the component is lacking a README.

I will work on one, but are you still unhappy with the prop names? What props would you expose?

Reply:
> Also, selectedItem and items very clearly convey the contract that the selectedItem should be an object from the items list.

I get that but the boat has sailed for a lot of components now, and I'd rather have users learn one way to use all of our controls rather than having each component with its own specifics.

> I will work on one, but are you still unhappy with the prop names? What props would you expose?

value, onChange and options being the important ones. 

>  I don't see how we gain from not exposing it.

we gain that if we ever change the implementation or anything, we're less likely to break backward compatibility. Adding new APIs anywhere in our codebase shouldn't be considered a "good thing", it's something to do with a lot of caution.

Reply:
>I get that but the boat has sailed for a lot of components now, and I'd rather have users learn one way to use all of our controls rather than having each component with its own specifics.

That's fair and I can definitely see value in that if we stick with it across all inputs. I updated the prop names in this PR: #18827.

>we gain that if we ever change the implementation or anything, we're less likely to break backward compatibility. Adding new APIs anywhere in our codebase shouldn't be considered a "good thing", it's something to do with a lot of caution.

Agreed, but in this case, it's something all select implementations could easily expose, so I wouldn't worry about backwards compatibility with it. I still haven't exposed it for now to be safe, we can do it when the need for it arises.